editor/src/core/shared/atom-with-pub-sub.ts

```diff
@@ -0,0 +1,76 @@
+import * as PubSub from 'pubsub-js'
+import * as React from 'react'
+
+export interface AtomWithPubSub<T> {
+  key: string
+  currentValue: T
+}
+
+const GlobalAtomMap: { [key: string]: AtomWithPubSub<any> } = {}
+
+export function atomWithPubSub<T>(options: { key: string; defaultValue: T }): AtomWithPubSub<T> {
```

Comment:
Why have you gone with `options: { key: string; defaultValue: T }` rather than two separate params here?